src/2_session.js

```diff
@@ -43,6 +45,37 @@ session.update = function(storage, newData) {
 		return;
 	}
 	var currentData = session.get(storage) || {};
-	var data = utils.merge(currentData, newData);
-	storage.set('branch_session', goog.json.serialize(data));
+	var data = goog.json.serialize(utils.merge(currentData, newData));
+	data = encodeBFPs(data);
+	storage.set('branch_session', data);
 };
+
+/**
+ * Encodes BFP in data object with Base64 encoding.
+ * BFP is supposed to be Base64 encoded when stored in local storage/cookie.
+ * @param {Object} data 
+ */
+function encodeBFPs(data){
+	if(data && !utils.isBase64Encoded(data["browser_fingerprint_id"])){
+		data["browser_fingerprint_id"] = btoa(data["browser_fingerprint_id"]);
```

Comment:
was the decision to still keep the keys with the same name?

Reply:
On the email thread between Alex, Aaron, You and I - there is no mention about changing the key. My understanding is that since all of this is strictly for obfuscation reason than anything else - we were not planning on encrypting the key.